### PR TITLE
Update static and async keyword recommenders

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to #39606

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/StaticKeywordRecommender.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 context.TargetToken.IsUsingKeywordInUsingDirective() ||
                 IsValidContextForType(context, cancellationToken) ||
                 IsValidContextForMember(context, cancellationToken) ||
+                context.SyntaxTree.IsLambdaDeclarationContext(position, otherModifier: SyntaxKind.AsyncKeyword, cancellationToken) ||
                 context.SyntaxTree.IsLocalFunctionDeclarationContext(position, s_validLocalFunctionModifiers, cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/StaticKeywordRecommender.cs:70]: order this the same as in the async recommender (or vice versa).

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs [hunk 2]
```diff
@@ -318,6 +318,42 @@ static bool IsGlobalAttributeList(AttributeListSyntax attributeList)
...
+            }
+
+            var modifierTokens = syntaxTree.GetPrecedingModifiers(position, token, out var beforeModifiersPosition);
+            if (modifierTokens.Count == 1 && modifierTokens.Contains(otherModifier))
+            {
+                if (token.HasMatchingText(SyntaxKind.AsyncKeyword))
+                {
+                    // second appearance of "async" not followed by modifier: treat as parameter name
+                    if (syntaxTree.GetPrecedingModifiers(token.SpanStart, token)
+                        .Contains(SyntaxKind.AsyncKeyword))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs:342]:
```suggestion
                    if (syntaxTree.GetPrecedingModifiers(token.SpanStart, token).Contains(SyntaxKind.AsyncKeyword))
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs [hunk 3]
```diff
@@ -318,6 +318,42 @@ static bool IsGlobalAttributeList(AttributeListSyntax attributeList)
...
+            {
+                return true;
+            }
+
+            var modifierTokens = syntaxTree.GetPrecedingModifiers(position, token, out var beforeModifiersPosition);
+            if (modifierTokens.Count == 1 && modifierTokens.Contains(otherModifier))
+            {
+                if (token.HasMatchingText(SyntaxKind.AsyncKeyword))
+                {
+                    // second appearance of "async" not followed by modifier: treat as parameter name
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs:340]: i really could not care less about thsi case :)

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/AsyncKeywordRecommender.cs [hunk 4]
```diff
@@ -25,18 +25,14 @@ public AsyncKeywordRecommender()
...
-
             if (context.TargetToken.IsKindOrHasMatchingText(SyntaxKind.PartialKeyword))
             {
                 return false;
             }
 
             return InMemberDeclarationContext(position, context, cancellationToken)
-                || context.SyntaxTree.IsLocalFunctionDeclarationContext(position, s_validLocalFunctionModifiers, cancellationToken);
+                || context.SyntaxTree.IsLocalFunctionDeclarationContext(position, s_validLocalFunctionModifiers, cancellationToken)
+                || context.SyntaxTree.IsLambdaDeclarationContext(position, otherModifier: SyntaxKind.StaticKeyword, cancellationToken);
```
Comment (RikkiGibson) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/AsyncKeywordRecommender.cs:35]:
```suggestion
                || context.SyntaxTree.IsLambdaDeclarationContext(position, otherModifier: SyntaxKind.StaticKeyword, cancellationToken)
                || context.SyntaxTree.IsLocalFunctionDeclarationContext(position, s_validLocalFunctionModifiers, cancellationToken);
```

**Pull request merged**